extensions/script-libraries/semver/semver.livecodescript

```diff
@@ -0,0 +1,726 @@
+
+A "version" is described by the v2.0.0 specification found at http://semver.org/.
+
+A leading `=` or `v` character is stripped off and ignored.
+
+## Ranges
+
+A version range is a set of comparators which specify versions that satisfy the range.
+
+A comparator is composed of an operator and a version. The set of primitive operators is:
+
+- `<` Less than
+- `<=` Less than or equal to
+- `>` Greater than
+- `>=` Greater than or equal to
+- `=` Equal. If no operator is specified, then equality is assumed, so this operator is
+optional, but MAY be included.
+
+For example, the comparator `>=1.2.7` would match the versions `1.2.7`, `1.2.8`, `2.5.3`,
+and `1.3.9`, but not the versions `1.2.6` or `1.1.0`.
+
+Comparators can be joined by whitespace to form a comparator set, which is satisfied by
+the intersection of all of the comparators it includes.
+
+A range is composed of one or more comparator sets, joined by ||. A version matches a
+range if and only if every comparator in at least one of the ||-separated comparator
+sets is satisfied by the version.
+
+For example, the range `>=1.2.7 <1.3.0` would match the versions `1.2.7`, `1.2.8`, and
+`1.2.99`, but not the versions `1.2.6`, `1.3.0`, or `1.1.0`.
+
+The range `1.2.7 || >=1.2.9 <2.0.0` would match the versions `1.2.7`, `1.2.9`, and `1.4.6`,
+but not the versions `1.2.8` or `2.0.0`.
+
+## Prerelease Tags
+
+If a version has a prerelease tag (for example, `1.2.3-alpha.3`) then it will only be
+allowed to satisfy comparator sets if at least one comparator with the same 
+[major, minor, patch] tuple also has a prerelease tag.
+
+For example, the range `>1.2.3-alpha.3` would be allowed to match the version `1.2.3-alpha.7`,
+but it would not be satisfied by `3.4.5-alpha.9`, even though `3.4.5-alpha.9` is technically 
+*greater than* `1.2.3-alpha.3` according to the SemVer sort rules. The version range only
+accepts prerelease tags on the `1.2.3` version. The version `3.4.5` would satisfy the range,
+because it does not have a prerelease flag, and `3.4.5` is greater than `1.2.3-alpha.7`.
+
+The purpose for this behavior is twofold. First, prerelease versions frequently are updated
+very quickly, and contain many breaking changes that are (by the author's design) not yet
+fit for public consumption. Therefore, by default, they are excluded from range matching
+semantics.
+
+Second, a user who has opted into using a prerelease version has clearly indicated the intent
+to use that specific set of alpha/beta/rc versions. By including a prerelease tag in the range,
+the user is indicating that they are aware of the risk. However, it is still not appropriate
+to assume that they have opted into taking a similar risk on the next set of prerelease
+versions.
+
+## Advanced Range Syntax
+
+Advanced range syntax desugars to primitive comparators in deterministic ways.
+
+Advanced ranges may be combined in the same way as primitive comparators using white space
+or ||.
+
+## Hyphen Ranges X.Y.Z - A.B.C
```

Comment:
These should probably be `###` headings shouldn't they? Aren't they sub-headings of advanced?